### PR TITLE
Add Michelson hooks

### PR DESCRIPTION
**Pull request by sskeirik**
First phase of the plugin update.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -10,3 +10,6 @@
 	path = deps/secp256k1
 	url = https://github.com/bitcoin-core/secp256k1
 	ignore = untracked
+[submodule "deps/cryptopp"]
+	path = deps/cryptopp
+	url = https://github.com/weidai11/cryptopp
```
Comment (ehildenb) [inline .gitmodules:15]:
```suggestion
	url = https://github.com/weidai11/cryptopp
	ignore = untracked
```

**Comment by ehildenb** [comment]
Please open PRs into relevant subrepos and link them here @sskeirik .

**Comment by sskeirik** [comment]
Finally coming back to this. Since it has been awhile, it is probably helpful to look at the overall plan here: #102. The changes in this PR represent the first phase of the plan laid out in that issue.

The linked kevm PR updating the plugin to this commit follows below:

https://github.com/kframework/evm-semantics/pull/807

**Comment by sskeirik** [comment]
@ehildenb kevm build passed, so this should be ready to review now. Let me know if you have any questions/comments. Thanks!

**Comment by ehildenb** [comment]
This LGTM, but I would like @dwightguth 's review as well.

**Pull request merged**